### PR TITLE
GH-103804: Add test for dis.findlinestarts

### LINKED ISSUE #103804: Add missing tests to the dis module
Improve test coverage for the dis module i.e `cpython/lib/dis.py`, tests are in `cpython/Lib/tests/test_dis.py`. Missing tests include:

- `dis.disco`
- `dis.findlinestarts`

### Linked PRs
* gh-103806
* gh-103901

### PR DESCRIPTION
**Pull request by jkchandalia**
PR to test dis.findlinestarts.

Fixes #103804.

* Issue: gh-103804

### CODE REVIEW

File: Lib/test/test_dis.py [hunk 1]
```diff
@@ -1935,6 +1935,15 @@ def test_findlabels(self):
...
         self.assertEqual(sorted(labels), sorted(jumps))
 
+    def test_findlinestarts(self):
+        def func():
+            pass
+        code = func.__code__
+        linestarts = list(dis.findlinestarts(code))
+        offsets = [linestart[0] for linestart in linestarts]
+        true_offsets = [0, 2]
+        self.assertEqual(offsets, true_offsets)
```
Comment (iritkatriel) [inline Lib/test/test_dis.py]:
```suggestion
        self.assertEqual(offsets, [0, 2])
```

File: Lib/test/test_dis.py [hunk 2]
```diff
@@ -1935,6 +1935,15 @@ def test_findlabels(self):
 
         self.assertEqual(sorted(labels), sorted(jumps))
 
+    def test_findlinestarts(self):
+        def func():
+            pass
+        code = func.__code__
+        linestarts = list(dis.findlinestarts(code))
+        offsets = [linestart[0] for linestart in linestarts]
```
Comment (iritkatriel) [inline Lib/test/test_dis.py]:
```suggestion
        offsets = [linestart[0] for linestart in dis.findlinestarts(code)]
```

Comment (iritkatriel) [reply Lib/test/test_dis.py]: (This avoids one list creation)

File: Lib/test/test_dis.py [hunk 3]
```diff
@@ -1935,6 +1935,15 @@ def test_findlabels(self):
 
         self.assertEqual(sorted(labels), sorted(jumps))
 
+    def test_findlinestarts(self):
+        def func():
+            pass
```
Comment (iritkatriel) [inline Lib/test/test_dis.py:1940]:
```suggestion
            pass
            
```

**Review by iritkatriel — APPROVED** [review]

**Review by iritkatriel — APPROVED** [review]

**Comment by nanjekyejoannah** [comment]
Congrats @jkchandalia for your first PR

**Review by mblahay — COMMENTED** [review]
This test appears to be accurate.

**Pull request merged**